### PR TITLE
Gorm Module - Separation of transactional gorm from standard gorm

### PR DESCRIPTION
**Pull request by wariosolis**
Changes in the module to use the database directly or through transactions Begin, Commit, Rollback

# Controller

```
type GormController struct {
	revel.Controller
	Txn *gorm.DB
}

func (c *GormController) Begin() revel.Result {
	c.Txn = DB
	...
}

```

# Transactional Controller

```
type GormTransactionController struct {
	revel.Controller
	Txn *gorm.DB
}


// Begin GormTransactionController to connect db
func (c *GormTransactionController) Begin() revel.Result {
	...
}

// Commit database transaction
func (c *GormTransactionController) Commit() revel.Result {
	...
}

// Rollback transaction
func (c *GormTransactionController) Rollback() revel.Result {
	...
}

```

Example: https://github.com/revel/examples/pull/30

### CODE REVIEW

File: gorm/app/gorm.go [hunk 1]
```diff
@@ -142,9 +175,9 @@ func init() {
 	revel.OnAppStart(func() {
 		if revel.Config.BoolDefault("db.autoinit", false) {
 			InitDB()
-			revel.InterceptMethod((*GormController).Begin, revel.BEFORE)
-			revel.InterceptMethod((*GormController).Commit, revel.AFTER)
-			revel.InterceptMethod((*GormController).Rollback, revel.FINALLY)
+			//revel.InterceptMethod((*GormController).Begin, revel.BEFORE)
+			//revel.InterceptMethod((*GormController).Commit, revel.AFTER)
+			//revel.InterceptMethod((*GormController).Rollback, revel.FINALLY)
```
Comment (notzippy) [inline gorm/app/gorm.go]: Why are these commented out now ?

Comment (wariosolis) [inline gorm/app/gorm.go]: I'm sorry, it was my mistake.

File: gorm/README.md [hunk 2]
```diff
@@ -27,3 +27,44 @@ db.password=dbpassword
...
+}
+
+func (c *GormController) Begin() revel.Result {
+	c.Txn = DB
+	...
+}
+
+```
+
+# Transaccional Controller
```
Comment (brendensoares) [inline gorm/README.md]: spelling

**Comment by wariosolis** [comment]
Fixed

File: gorm/README.md [hunk 3]
```diff
@@ -27,3 +27,44 @@ db.password=dbpassword
 
 ```
 
+
+# Controller
+
+```
+type GormController struct {
+	revel.Controller
```
Comment (notzippy) [inline gorm/README.md]: Why are you embedding revel.Controller in this manner ? If this type was in the controllers folder you could embed the type directly inside another controller like
```
type MyController struct {
  gorm.GormController
}

Comment (wariosolis) [inline gorm/README.md]: The readme specifies how it works. Not how to use it

The way of use is to do the following:

```
type MyController struct {
    gorm.GormController
}
```

**Comment by notzippy** [comment]
@wariosolis I am closing this in favor of #51. That PR has the transactional controller and it has refactored the controller to the folder `controllers`, which I believe was something that you had mentioned wanting to do in this PR. I am closing this , but feel free to reopen if I am missing something.
thanks

**Pull request closed without merging**